### DISCUSSION TITLE
Docker Swarm

### DISCUSSION THREAD

**Discussion by fagurto** [appwrite/appwrite#4440] [Q&A]
hi guys!!! just wondering if you could give some help .

the docker-compose file provided in https://appwrite.io/docker-compose.yml  can be run as a docker swarm stack ??? , would be great to have some examples  of appwrite running on swarm or k8s.

regards!

**Comment by damascenaluiz** [comment]
I managed to make it work.
Some important details were:

1 - Create an external overlay network. See: [https://docs.docker.com/network/overlay/]()
2 - Open the necessary ports for communication between hosts participating in the swarm, only between them.
3 - Rename all host addresses in the configuration .yml file by adding the name of your stack plus an underscore, for example: `_APP_REDIS_HOST = redis` -> `_APP_REDIS_HOST = my-stack_redis`.
4 - Run with `docker stack deploy --compose-file docker-compose.yml my-stack`

This example also contains the folder of uploads on nfs, which worked well, see that AV disabled by the amount of memory required, valid if it is for testing.

Version 0.7, I was unable to do it with swarm, I will still do more tests.

[testing_docker-compose.txt](https://github.com/appwrite/appwrite/files/5983492/testing_docker-compose.txt)

**Comment by stnguyen90** [comment]
The best resource to follow on deploying Appwrite with Docker Swarm is our [Docker Swarm Integration](https://dev.to/appwrite/30daysofappwrite-docker-swarm-integration-2io9) article.

**Reply by GFoley83** [reply]
Hey @stnguyen90 are those articles still ok or out of date now? 

Also, are there any docs you know of for deploying and hosting in AWS at all? 
Eg using RDS for MariaDb, Redis etc

Thanks!
